lib/ecl/ecl_grid.cpp

```diff
@@ -1871,7 +1871,29 @@ static void ecl_grid_set_active_index(ecl_grid_type * ecl_grid) {
   } else {
     /* --- More involved path in the case of coarsening groups. --- */
 
-    /* 1: Go through all the cells and set the active index. In the
+    /* 1: Reset cell active_index across the entire grid.
+          In the involved path ecl_coarse_cell_update_index() only updates
+          the coarse cells' active_index if it is -1. */
+    for (global_index = 0; global_index < ecl_grid->size; global_index++) {
+        ecl_cell_type * cell = ecl_grid_get_cell(ecl_grid, global_index);
+        if (cell->coarse_group == COARSE_GROUP_NONE)
+        {
+            if (cell->active & CELL_ACTIVE_MATRIX)
+            {
+                cell->active_index[MATRIX_INDEX] = -1;
+            }
+
+            if (cell->active & CELL_ACTIVE_FRACTURE)
+            {
+                cell->active_index[FRACTURE_INDEX] = -1;
+            }
```

Comment:
Unless I missed something, the `cell->active_index` and `active_fracture_index` are just overwritten further down (lines 1906 and 1911 at the time of this review), with no checks about their previous values. I suspect we can remove the entire `if` and keep just the lines in the `else` block

Reply:
that looks reasonable